src/sentry/integrations/bitbucket/integration.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import absolute_import
+
+from sentry.integrations import IntegrationProvider, IntegrationMetadata
+from sentry.pipeline import NestedPipelineView
+from sentry.identity.pipeline import IdentityProviderPipeline
+from django.utils.translation import ugettext_lazy as _
+from sentry.utils.http import absolute_uri
+from sentry.models import Integration
+
+DESCRIPTION = """
+Bitbucket for Sentry.io
+"""
+metadata = IntegrationMetadata(
+    description=DESCRIPTION.strip(),
+    author='The Sentry Team',
+    noun=_('Bitbucket Account'),
+    issue_url='https://github.com/getsentry/sentry/issues/new?title=Bitbucket%20Integration:%20&labels=Component%3A%20Integrations',
+    source_url='https://github.com/getsentry/sentry/tree/master/src/sentry/integrations/bitbucket',
+    aspects={},
+)
+
+
+class BitbucketIntegrationProvider(IntegrationProvider):
+    key = 'bitbucket'
+    name = 'Bitbucket'
+    metadata = metadata
+
+    def get_pipeline_views(self):
+        identity_pipeline_config = {
+            'redirect_url': absolute_uri('/extensions/bitbucket/setup/'),
+        }
+        identity_pipeline_view = NestedPipelineView(
+            bind_key='identity',
+            provider_key='bitbucket',
+            pipeline_cls=IdentityProviderPipeline,
+            config=identity_pipeline_config,
+        )
+
+        return [
+            identity_pipeline_view,
+        ]
+
+    def build_integration(self, state):
+        # TODO(LB): Add verification for clientKey
+        if state.get('publicKey'):
+            principal_data = state['principal']
+            user_data = state['user']
+            return {
+                'provider': 'bitbucket',
+                'external_id': state['clientKey'],
+                'name': principal_data['username'],
+                'metadata': {
+                    'public_key': state['publicKey'],
+                    'shared_secret': state['sharedSecret'],
+                    'base_url': state['baseApiUrl'],
+                    'domain_name': principal_data['links']['html']['href'].replace('https://', ''),
+                    'icon': principal_data['links']['avatar']['href'],
+                },
+                'user_identity': {
+                    'type': 'bitbucket',
+                    'name': user_data['username'],
+                    'display_name': user_data['display_name'],
+                    'account_id': user_data['account_id'],
+                    'icon': user_data['links']['avatar']['href'],
+                }
```

Comment:
I think we probably don't want to include `user_identity` at all, since it looks like we don't get back any kind of `external_id` for the user right?

Reply:
@EvanPurkhiser I'm a bit confused. We do in fact get the user information. the whole the `external_id` is either the `username` or the `UUID`. Im not super familiar with a lot of these concepts, so perhaps I'm saving the wrong fields. Lets meet up and go over what would be best to save! 

But we definitely do have user information. Is there a specific piece of information for the user you believe must be available to make it worthwhile to save?

Reply:
I'm going to put the uuid in there I think because when you authorize with a team, you don't get `account_id`. I'm not sure what `account_id` is used for, but I think `uuid` will be generic and we can make api calls with it.